src/fips.c

```diff
@@ -330,41 +353,39 @@ static int compute_file_hmac(const char *path, void **buf, size_t *hmaclen)
 	if (!mdctx)
 		goto end;
 
-	fp = fopen(path, "r");
-	if (fp == NULL)
+	fdata = mmap_file(path, &fdata_stat);
+	if (fdata == NULL)
 		goto end;
 
-	if (EVP_DigestSignInit(mdctx, NULL, EVP_sha256(), NULL, pkey) <= 0)
+	if (EVP_DigestSignInit(mdctx, NULL, EVP_sha256(), NULL, pkey) != 1)
 		goto end;
 
-	while ((len = fread(rbuf, 1, sizeof(rbuf), fp)) != 0) {
-		if (EVP_DigestSignUpdate(mdctx, rbuf, len) <= 0) {
-			goto end;
-		}
-	}
+	if (EVP_DigestSignUpdate(mdctx, fdata, fdata_stat.st_size) != 1)
+		goto end;
 
-	hlen = sizeof(rbuf);
-	if (EVP_DigestSignFinal(mdctx, rbuf, &hlen) <= 0)
+	if (EVP_DigestSignFinal(mdctx, tmp, &tmp_len) != 1)
```

Comment:
You may actually use EVP_DigestSign() instead of EVP_DigestSignUpdate() followed by EVP_DigestSignFinal().

Reply:
Ok, I'll change it.